packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -201,11 +202,38 @@ export function createAlgoliaInsightsPlugin(
       insightsClient('addAlgoliaAgent', 'insights-plugin');
 
       setInsightsContext();
-      insightsClient('onUserTokenChange', setInsightsContext);
+
+      // Handles user token changes
+      insightsClient('onUserTokenChange', (userToken) => {
+        setInsightsContext(userToken);
+      });
```

Comment:
This can be inlined as it used to be, but not a blocker.

```suggestion
      insightsClient('onUserTokenChange', setInsightsContext);
```